File: Backend/src/main/java/com/vidyasahay/vidyasahay/service/BankService.java
package com.vidyasahay.vidyasahay.service;

import com.vidyasahay.vidyasahay.dto.response.BankDetailedResponse;
import com.vidyasahay.vidyasahay.dto.response.BankSummaryResponse;
import com.vidyasahay.vidyasahay.dto.request.CreateBankRequest;
import com.vidyasahay.vidyasahay.dto.request.UpdateBankRequest;

import java.util.List;
import java.util.UUID;

public interface BankService {

    void createBank(CreateBankRequest request);

    List<BankSummaryResponse> getAllBanks();

    BankDetailedResponse getBankById(UUID bankId);

    void updateBank(UUID bankId, UpdateBankRequest request);
}

File: Frontend/src/app/core/services/bank.service.ts
import { Injectable } from '@angular/core';
import { BankSummary} from '../models/summaryResponse.dto';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/env'
import { CreateBankRequest } from '../models/create-bank.dto';
import { BankDetails, UpdateBankRequest } from '../models/bank-details.dto';

@Injectable({
  providedIn: 'root'
})
export class BankService {

  private readonly API_URL = `${environment.apiUrl}/bank`;

  constructor(private http: HttpClient) { }

  getBanks(): Observable<BankSummary[]> {
    return this.http.get<BankSummary[]>(`${this.API_URL}/all`);
  }

  createBank(request: CreateBankRequest): Observable<void> {
    return this.http.post<void>(this.API_URL, request);
  }

  getBankById(bankId: string): Observable<BankDetails> {
    return this.http.get<BankDetails>(`${this.API_URL}/${bankId}`);
  }

  updateBank(bankId: string, request: UpdateBankRequest): Observable<void> {
    return this.http.patch<void>(`${this.API_URL}/${bankId}`, request);
  }
}
